### PR TITLE
raiyan: run react app using docker.

### PR DESCRIPTION
**Pull request by raiyan-hosn**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
Added docker to run React todo app.
## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: todo-react-app/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+
+FROM node:alpine
+# alpine is a light-weight linux distro with nodejs. We use it for better permormance.
+
+WORKDIR /app
+# switching to a new working directory /app from the container's root. 
+
+COPY . .
```
Comment (musfique-m) [inline todo-react-app/Dockerfile:8]: Please copy the files after we install packages using npm install. There will be less frequent changes in package-lock files and so we can take advantages of docker caching from here.
Our program files will change more frequently, so every time we build an image the COPY instruction will create a new layer. So caching will not be applied in that case. If you keep your instructions in this order docker will install the packages in every build.

**Review by musfique-m — CHANGES_REQUESTED** [review]
Please remove the other files that are not related to Docker from this PR.

File: todo-react-app/Dockerfile (hunk 1 above)
Comment (raiyan-hosn) [reply todo-react-app/Dockerfile]: Resolved.

File: todo-react-app/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# alpine is a light-weight linux distro with nodejs. We use it for better permormance.
+
+WORKDIR /app
+# switching to a new working directory /app from the container's root. 
+
+COPY package.json .
+# coping package.json to app/ dir.
+
+COPY package-lock.json .
+# coping package-lock.json to app/ dir.
```
Comment (musfique-m) [inline todo-react-app/Dockerfile:12]: Please reduce the copy operation into a single layer.

**Review by musfique-m — CHANGES_REQUESTED** [review]

File: todo-react-app/Dockerfile (hunk 2 above)
Comment (raiyan-hosn) [reply todo-react-app/Dockerfile]: Reduced copy operation.

**Review by musfique-m — APPROVED** [review]
Looks good.

**Pull request merged**